cdc/model/codec/codec.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codec
+
+import (
+	"encoding/binary"
+
+	timodel "github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tiflow/cdc/model"
+	codecv1 "github.com/pingcap/tiflow/cdc/model/codec/v1"
+	"github.com/tinylib/msgp/msgp"
+)
+
+const (
+	v1HeaderLength      int = 4
+	versionPrefixLength int = 2
+	versionFieldLength  int = 2
+
+	latestVersion uint16 = 2
+)
+
+// NOTE: why we need this?
+//
+// Before this logic is introduced, redo log is encoded into byte slice without a version field.
+// This makes it hard to extend in future.
+// However in the old format (i.e. v1 format), the first 5 bytes are always same, which can be
+// confirmed in v1/codec_gen.go. So we reuse those bytes, and add a version field in them.
+var (
+	v1Header      [v1HeaderLength]byte      = [...]byte{0x83, 0xa3, 0x72, 0x6f}
+	versionPrefix [versionPrefixLength]byte = [...]byte{0xff, 0xff}
```

Comment:
```suggestion
	versionPrefix = [versionPrefixLength]byte{0xff, 0xff}
```

Reply:
where the header's value come from?

Reply:
v1Header comes from v1/codec_gen.go. versionPrefix is defined by us.